src/utils/auth/auth.service.ts

```diff
@@ -12,12 +13,24 @@ export class AuthService {
     const token = (await jwt_decode(encodedToken)) as any;
 
     if (!token.email) throw new AuthenticationException('Token email missing!');
-    const knownUser = await this.userService.getUserWithGroups(token.email);
+    let knownUser = await this.userService.getUserWithGroups(token.email);
 
-    if (!knownUser)
-      throw new AuthenticationException(
-        `No user with email ${token.email} found!`
-      );
+    if (!knownUser) {
+      // User does not yet have a profile on the platform, create one
+      const userData = new UserInput();
+      userData.email = token.email;
```

Comment:
This shouldn't be here - the business logic building the new object should be in the factory creating the object, validating the DTO should be using decorators in the DTO (Input) class. You can also decorate the fields in the entity class with default values - that will make init consistent across the application.

Reply:
Agreed - added separate issue (#786) to update this code to throw a specific exception that can be picked up by the client.